src/test/java/software/amazon/cloudformation/LambdaWrapperTest.java

```diff
@@ -190,6 +190,36 @@ public void invokeHandler_nullResponse_returnsFailure(final String requestDataPa
         }
     }
 
+    @Test
+    public void invokeHandler_SchemaFailureOnNestedProperties() throws IOException {
+        // use actual validator to verify behaviour
+        final WrapperOverride wrapper = new WrapperOverride(providerLoggingCredentialsProvider, platformEventsLogger,
+                                                            providerEventsLogger, providerMetricsPublisher, new Validator() {
+                                                            }, httpClient);
+
+        wrapper.setTransformResponse(resourceHandlerRequest);
+
+        try (final InputStream in = loadRequestStream("create.request.with-extraneous-model-object.json");
+            final OutputStream out = new ByteArrayOutputStream()) {
+            final Context context = getLambdaContext();
+
+            wrapper.handleRequest(in, out, context);
+            // validation failure metric should be published but no others
+            verify(providerMetricsPublisher).publishExceptionMetric(any(Instant.class), eq(Action.CREATE), any(Exception.class),
+                any(HandlerErrorCode.class));
+
+            // all metrics should be published, even for a single invocation
+            verify(providerMetricsPublisher).publishInvocationMetric(any(Instant.class), eq(Action.CREATE));
+
+            // verify initialiseRuntime was called and initialised dependencies
+            verifyInitialiseRuntime();
+
+            // verify output response
+            verifyHandlerResponse(out, ProgressEvent.<TestModel, TestContext>builder().errorCode(HandlerErrorCode.InvalidRequest)
```

Comment:
The InvalidRequest looks right. But how does it match with previous `TerminalException`?

Reply:
it matches bc terminal exception was used just to create throwable but handler error code is set to `InvalidRequest`:

```java
ProgressEvent.defaultFailureHandler(
    new TerminalException("Model validation failed caused by invalid input provided", e),
    HandlerErrorCode.InvalidRequest);
```